index-pingbacks.js

```diff
@@ -0,0 +1,104 @@
+import http from "node:http";
+import https from "node:https";
+import log from "lambda-log";
+import { decodeRecords } from "./lib/decoder.js";
+import * as pingurl from "./lib/pingurl.js";
+import * as timestamp from "./lib/timestamp.js";
+import * as urlutil from "./lib/urlutil.js";
+
+/**
+ * GET impression pingbacks and POST impression increments to dovetail router,
+ * so it has up to date counts of what has served.
+ */
+export const handler = async (event) => {
+  const records = await decodeRecords(event);
+
+  // we only care about non-dup impressions here
+  const imps = records
+    .filter((r) => r.type === "postbytes")
+    .flatMap((r) => (r.impressions || []).map((i) => ({ ...r, ...i })))
+    .filter((r) => !r.isDuplicate);
+  const pings = imps.flatMap((r) => (r.pings || []).map((pingUrl) => ({ ...r, pingUrl })));
+
+  const info = { records: records.length, pingbacks: pings.length, increments: imps.length };
+  log.info("Starting Pingbacks", info);
+
+  // manually create agents, so we can cleanup sockets (TODO: is this still needed?)
+  const agents = { http: new http.Agent(), https: new https.Agent() };
+
+  // run pingbacks
+  const pingRes = await Promise.all(pings.map((rec) => runPingback(rec, agents)));
+  const pingCount = pingRes.filter((r) => r).length;
+  const pingFail = pingRes.filter((r) => !r).length;
+
+  // run increments
+  const hosts = (process.env.DOVETAIL_ROUTER_HOSTS || "").split(",");
+  const tokens = (process.env.DOVETAIL_ROUTER_API_TOKENS || "").split(",");
+  const incrs = formatIncrements(imps);
+  await Promise.all(hosts.map((h, i) => runIncrement(h, tokens[i], incrs, agents)));
```

Comment:
Would it make sense for this to be a different lambda?
I mean, it's the same impression data, but sort of different purpose...